pwndbg/commands/heap.py

```diff
@@ -80,41 +87,59 @@ def bins(addr=None):
     fastbins = main_arena['fastbinsY']
     bins = main_arena['bins']
 
+    size_t_size = gdb.lookup_type('size_t').sizeof
     num_fastbins = int(fastbins.type.sizeof / fastbins.type.target().sizeof)
     num_bins = int(bins.type.sizeof / bins.type.target().sizeof)
-    fd_field_offset = 16
+    fd_field_offset = 2 * size_t_size
 
     print(underline(yellow('Fastbins')))
     for i in range(num_fastbins):
+        size = 2 * size_t_size * (i + 1)
         chain = pwndbg.chain.format(int(fastbins[i]), offset=fd_field_offset)
-        print(bold(str(i)) + ': ' + chain)
+        print(bold(str(size)) + ': ' + chain)
 
     # TODO: Print other bins
 
-@pwndbg.commands.Command
+@pwndbg.commands.ParsedCommand
 @pwndbg.commands.OnlyWhenRunning
 def top_chunk(addr=None):
     main_arena = get_main_arena(addr)
     if main_arena == None:
-        return
-
-    top = main_arena['top']
-    print(top)
+        heap_start, heap_end = get_heap_bounds()
+        if heap_start == None:
+            print(red('Could not find the heap'))
+            return
+
+        # If we don't know where the main_arena struct is, just iterate
+        # through all the heap objects until we hit the last one
+        last_addr = None
+        addr = heap_start
+        while addr < heap_end:
+            chunk = gdb.parse_and_eval('(struct malloc_chunk)*' + hex(addr))
+            size = int(chunk['size'])
+
+            # Clear the bottom 3 bits
+            size &= ~7
+
+            last_addr = addr
+            addr += size
+        print(pwndbg.color.get(last_addr))
+    else:
+        print(pwndbg.color.get(int(main_arena['top'])))
 
-@pwndbg.commands.Command
+@pwndbg.commands.ParsedCommand
 @pwndbg.commands.OnlyWhenRunning
 def malloc_chunk(addr):
-    if isinstance(addr, (long, int)):
-        addr = hex(addr)
+    if not isinstance(addr, int):
```

Comment:
I think that was true when you were calling `hex`, but now you're just calling `int`.  The code now is `if not isinstance(foo, int): foo = int(foo)`; the `if` is a noop.  In any case, it doesn't matter :)